diff --git a/indra/newview/viewer_manifest.py b/indra/newview/viewer_manifest.py
index 3a5fc43fee..c86b76158a 100755
--- a/indra/newview/viewer_manifest.py
+++ b/indra/newview/viewer_manifest.py
@@ -870,6 +870,7 @@ class Windows_x86_64_Manifest(ViewerManifest):
         if not self.is_packaging_viewer():
             self.package_file = "copied_deps"
 
+        if self.is_packaging_viewer(): self.path(src="../packages/lib/release/openvr_api.dll", dst="openvr_api.dll")
         self.fs_copy_windows_manifest( )
 
     def nsi_file_commands(self, install=True):
