--- fs-7.1.13-devtime-avx2/source/newview/llviewerdisplay.cpp	2025-08-03 12:31:15.000000000 -0400
+++ sgeo-minimal-vrmod.llviewerdisplay.cpp	2025-08-04 02:04:36.556753400 -0400
@@ -26,6 +26,11 @@
 
 #include "llviewerprecompiledheaders.h"
 
+//################################### P373R ######################################
+#include "llviewerVR.cpp"
+llviewerVR gVR;
+//################################### END P373R ##################################
+
 #include "llviewerdisplay.h"
 
 #include "fsyspath.h"
@@ -832,6 +837,12 @@
             gPipeline.toggleRenderType(LLPipeline::RENDER_TYPE_HUD_PARTICLES);
         }
 
+		//################################### P373R ######################################
+		sec:
+		gVR.ProcessVRCamera();
+		//################################### END P373R ##################################
+		
+
         stop_glerror();
         display_update_camera();
         stop_glerror();
@@ -1154,7 +1165,29 @@
         if (!for_snapshot)
         {
             render_ui();
-            swap();
+			//################################### P373R ######################################
+			gVR.vrDisplay();
+
+			//################################### END P373R ######################################
+			//swap();
+			//################################### P373R ######################################
+			if (gVR.leftEyeDesc.IsReady  && !gVR.rightEyeDesc.IsReady && gVR.eyeDistance() > 0)
+			{
+				goto sec;
+
+
+			}
+			if (!gVR.leftEyeDesc.IsReady  && !gVR.rightEyeDesc.IsReady)
+			{
+				//gVR.HandleInput();
+
+			}
+			if (!gVR.m_bVrActive)
+ 			swap();
+
+			//################################### END P373R ##################################
+			//swap();
+			
         }
 
 
@@ -1787,6 +1820,10 @@
     gUIProgram.bind();
     gGL.color4f(1, 1, 1, 1);
 
+	//################################### P373R ######################################
+	gVR.RenderControllerAxes();
+	//################################### END P373R ##################################
+
     // Coordinate axes
     // <FS:Ansariel> gSavedSettings replacement
     //if (gSavedSettings.getBOOL("ShowAxes"))
@@ -1939,6 +1976,9 @@
         gViewerWindow->draw();
     }
 
+	//################################### P373R ######################################
+	gVR.DrawCursors();
+	//################################### END P373R ##################################
 
 
     // reset current origin for font rendering, in case of tiling render
@@ -2024,6 +2064,9 @@
 
 void display_cleanup()
 {
+	//################################### P373R ######################################
+	gVR.vrStartup(TRUE);
+	//################################### END P373R ######################################
     gDisconnectedImagep = NULL;
 }
 
