--- snapshot/source/newview/llviewerregion.h	2025-10-27 15:35:27.000000000 -0400
+++ llviewerregion.h	2025-12-31 18:46:11.377548023 -0500
@@ -106,7 +106,7 @@
         NUM_PARTITIONS
     } eObjectPartitions;
 
-    typedef boost::signals2::signal<void(const LLUUID& region_id, LLViewerRegion* regionp)> caps_received_signal_t;
+    typedef boost::signals2::signal<void(const LLUUID& region_id, LLViewerRegion* const& regionp)> caps_received_signal_t;
 
     LLViewerRegion(const U64 &handle,
                    const LLHost &host,
